pkgs/stdenv/generic/make-derivation.nix

```
@@ -87,7 +87,46 @@ let
       args = rattrs (args // { inherit finalPackage overrideAttrs; });
       #              ^^^^
 
-      overrideAttrs = f0: makeDerivationExtensible (lib.extends (lib.toExtension f0) rattrs);
+      overrideAttrs =
+        f0:
+        let
+          extends' =
+            overlay: f:
+            (
+              final:
+              let
+                prev = f final;
+                thisOverlay = overlay final prev;
+                warnForBadVersionOverride = (
+                  thisOverlay ? version
+                  && !(thisOverlay ? src)
+                  && !thisOverlay.__intentionallyOverridingVersion or false
```

Should I add parens for clarity, or is that just a me thing?
```suggestion
                  && !(thisOverlay.__intentionallyOverridingVersion or false)
```

---

I always do for clarity's sake but as long as it's correct I don't really care.

---

I mean isn’t the expression evaluating to (literally) `!thisOverlay.__intentionallyOverridingVersion` or `false` if it’s unset? Or am I messing up the precedence?

Okay yeah since I’m not the only one with this opinion I’ll add parens.